### PR TITLE
[REVIEW] Add `decimal64Dtype` support in data-generator

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR adds `decimal64Dtype` support in the data-generator for the purpose of fuzz-testing. The changes also enable fuzz testing in parquet as this dtype is currently supported in parquet reader and writer only.

### CODE REVIEW

File: python/cudf/cudf/_fuzz_testing/utils.py [hunk 1]
```diff
@@ -114,6 +114,8 @@ def _generate_rand_meta(obj, dtypes_list, null_frequency_override=None):
             meta["value_type"] = random.choice(
                 list(cudf.utils.dtypes.ALL_TYPES - {"category"})
             )
+        elif dtype == "decimal64":
+            meta["max_precision"] = np.floor(np.log10(np.iinfo("int64").max))
```
Comment (shwina) [inline python/cudf/cudf/_fuzz_testing/utils.py:118]: The `Decimal64Dtype` exposes a `MAX_PRECISION` attribute - can we use that here instead?

File: python/cudf/cudf/tests/dataset_generator.py [hunk 2]
```diff
@@ -131,30 +131,47 @@ def _generate_column(column_params, num_rows):
...
+                    p=[
+                        column_params.null_frequency,
+                        1 - column_params.null_frequency,
+                    ],
+                )
+                if column_params.null_frequency > 0.0
+                else None,
+                size=num_rows,
+                safe=False,
+                type=arrow_type,
```
Comment (shwina) [inline python/cudf/cudf/tests/dataset_generator.py:173]: Can we avoid some code duplication here with just using `type=arrow_type if not isinstance(arrow_type, pa.lib.Decimal128Type) else None`, and then casting if necessary?

File: python/cudf/cudf/_fuzz_testing/parquet.py [hunk 3]
```diff
@@ -146,7 +146,7 @@ def generate_input(self):
                 # https://github.com/pandas-dev/pandas/issues/37327
                 - {"uint32"}
             )
-            dtypes_list.extend(["list"])
+            dtypes_list.extend(["list", "decimal64"])
```
Comment (rgsl888prabhu) [inline python/cudf/cudf/_fuzz_testing/parquet.py:149]: Just a suggestion

```suggestion
            dtypes_list = list(
                cudf.utils.dtypes.ALL_TYPES
                - {"category", "timedelta64[ns]", "datetime64[ns]"}
                # TODO: Remove uint32 below after this bug is fixed
                # https://github.com/pandas-dev/pandas/issues/37327
                - {"uint32"} | {"list", "decimal64"}
            )
```

File: python/cudf/cudf/tests/dataset_generator.py [hunk 4]
```diff
@@ -340,6 +357,22 @@ def rand_dataframe(dtypes_meta, rows, seed=random.randint(0, 2 ** 32 - 1)):
                     dtype=dtype,
                 )
             )
+        elif dtype == "decimal64":
+            max_precision = meta.get(
+                "max_precision", np.floor(np.log10(np.iinfo("int64").max))
+            )
```
Comment (vuule) [inline python/cudf/cudf/tests/dataset_generator.py:363]: https://github.com/rapidsai/cudf/pull/8107/files?diff=unified&w=1#r623159828
same here

File: python/cudf/cudf/tests/dataset_generator.py (hunk 2 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/dataset_generator.py]: Changed it 👍

**Review by vuule — APPROVED** [review]
FWIW, looks good to me :)

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
@gpucibot merge

**Pull request merged**